sdk/test/UnitTests/Custom/TestTools/Comparer.cs

```diff
@@ -51,6 +53,14 @@ private static void Compare(object x, object y, Type type)
             if (x == null && y == null)
                 return;
 
+            if (x != null && x.GetType().GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(ICollection<>)) && !AWSConfigs.InitializeCollections)
```

Comment:
The comparer should handle both possible boolean values for the `InitializeCollections` property.

Reply:
The other if statements handle when `InitializeCollections` is true, but I did simplify the if statement and the statement below.